endpoints/routing/handler.go

```diff
@@ -15,16 +15,37 @@ import (
 	"github.com/rs/cors"
 )
 
-func NewHandler(cfg config.Configuration, dataStore backends.Backend, appMetrics *metrics.Metrics) http.Handler {
+func NewAdminHandler(cfg config.Configuration, dataStore backends.Backend, appMetrics *metrics.Metrics) http.Handler {
 	router := httprouter.New()
-	router.GET("/", endpoints.Index)        //Default route handler
-	router.GET("/status", endpoints.Status) // Determines whether the server is ready for more traffic.
-	router.POST("/cache", decorators.MonitorHttp(endpoints.NewPutHandler(dataStore, cfg.RequestLimits.MaxNumValues, cfg.RequestLimits.AllowSettingKeys), appMetrics, decorators.PostMethod))
-	router.GET("/cache", decorators.MonitorHttp(endpoints.NewGetHandler(dataStore, cfg.RequestLimits.AllowSettingKeys), appMetrics, decorators.GetMethod))
+	router = addReadOnlyRoutes(cfg, dataStore, appMetrics, router)
+	router = addWriteRoutes(cfg, dataStore, appMetrics, router)
+	return router
+}
+
+func NewPublicHandler(cfg config.Configuration, dataStore backends.Backend, appMetrics *metrics.Metrics) http.Handler {
+	router := httprouter.New()
+	router = addReadOnlyRoutes(cfg, dataStore, appMetrics, router)
+	if cfg.PublicWriteEnabled {
+		router = addWriteRoutes(cfg, dataStore, appMetrics, router)
+	}
 
 	handler := handleCors(router)
 	handler = handleRateLimiting(handler, cfg.RateLimiting)
-	return handler
+	return router
+}
+
+func addReadOnlyRoutes(cfg config.Configuration, dataStore backends.Backend, appMetrics *metrics.Metrics, router *httprouter.Router) *httprouter.Router {
```

Comment:
Nitpick: Consider renaming to `addReadRoutes` to match `addWriteRoutes`.

Reply:
Makes sense. Done.